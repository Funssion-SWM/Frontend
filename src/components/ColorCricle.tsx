type Props = {
  color: string;
  selected: string;
  onClick: (color: string) => void;
};

export default function ColorCricle({ color, selected, onClick }: Props) {
  return (
    <div
      onClick={() => onClick(color)}
      className={`w-6 h-6 rounded-full cursor-pointer ${color} ${
        selected === color && 'border-2 border-black'
      }`}
    ></div>
  );
}
