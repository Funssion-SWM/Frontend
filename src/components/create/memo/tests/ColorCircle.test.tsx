import renderer from 'react-test-renderer';
import ColorCricle from '../ColorCircle';
import { mock } from 'node:test';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ColorCircle', () => {
  it('renders correctly when selected', () => {
    const component = renderer.create(
      <ColorCricle color={'yellow'} selected={'yellow'} onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when not selected', () => {
    const component = renderer.create(
      <ColorCricle color={'yellow'} selected={'green'} onClick={() => {}} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('clicks correctly', async () => {
    const mockCallBack = mock.fn();

    render(
      <ColorCricle
        color={'yellow'}
        selected={'yellow'}
        onClick={mockCallBack}
      />
    );

    const circle = screen.getByText('âœ“');
    await userEvent.click(circle);

    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
