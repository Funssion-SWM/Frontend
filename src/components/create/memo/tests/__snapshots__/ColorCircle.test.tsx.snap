// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCircle renders correctly when not selected 1`] = `
<div
  className="w-6 h-6 rounded-full cursor-pointer border-[1px] border-black bg-memo-yellow "
  onClick={[Function]}
/>
`;

exports[`ColorCircle renders correctly when selected 1`] = `
<div
  className="w-6 h-6 rounded-full cursor-pointer border-[1px] border-black bg-memo-yellow border-[2px] border-soma-blue-50 text-soma-blue-50 text-center"
  onClick={[Function]}
>
  âœ“
</div>
`;
